### ISSUE TITLE
SC2154 unassigned variable not reported in array assignments

### ISSUE THREAD

**Issue by kvantour** [koalaman/shellcheck#1969]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2154 
- My shellcheck version (`shellcheck --version` or "online"): 0.7.1
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/usr/bin/env bash
a=()
a+=( "$foo" )

```

#### Here's what shellcheck currently says:
No issues detected!

#### Here's what I wanted or expected to see:
SC2154: foo is referenced but not assigned.

**Comment by brother** [comment]
Online shellcheck is not affected. It seems to have been taken care of in a later version.

**Comment by kvantour** [comment]
Still there in the current version of shellcheck (version 0.9.0)

**Comment by brother** [comment]
Not in HEAD though. See https://shellcheck.net -  if possivble upgrade to a local build from HEAD.

**Comment by ruricolist** [comment]
This problem still exists (or has recurred) in the version running on shellcheck.net.

**Comment by e-kwsm** [comment]
shellcheck 0.11.0 and the online version issue SC2154.
